pkg/filter/filter.go

```diff
@@ -114,35 +116,55 @@ func New(caseSensitive bool, rules *Rules) *Filter {
 	f.c = &cache{
 		items: make(map[string]ActionType),
 	}
-	f.genRegexMap()
-	return f
+	err := f.genRegexMap()
+	return f, err
```

Comment:
if `err != nil`, the behavior of `f` is unpredictable, should we need to return `nil` rather than `f`?